### PR TITLE
[Podfile DSL] Document how subspecs can be included

### PR DESCRIPTION
**Pull request by kylef**

### CODE REVIEW

File: lib/cocoapods-core/podfile/dsl.rb [hunk 1]
```diff
@@ -112,6 +112,19 @@ module DSL
       #
       # ------
       #
+      # ### Subspecs
+      #
+      # By default, all of a Pod's default subspecs are installed, unless
```
Comment (segiddins) [inline lib/cocoapods-core/podfile/dsl.rb]: not true, only the default subspecs

File: lib/cocoapods-core/podfile/dsl.rb [hunk 2]
```diff
@@ -112,6 +112,19 @@ module DSL
...
+      # you specifically specify a subspec as follows:
+      #
+      #     pod 'QueryKit/Attribute'
+      #
+      # You may specify a collection of subspecs to be installed as follows:
+      #
+      #     pod 'QueryKit', :subspecs => ['Attribute', 'QuerySet']
```
Comment (segiddins) [inline lib/cocoapods-core/podfile/dsl.rb]: should this mention that it also means installing the root spec?

File: lib/cocoapods-core/podfile/dsl.rb (hunk 1 above)
Comment (orta) [inline lib/cocoapods-core/podfile/dsl.rb]: Wording is a bit strange in general ( default used twice )

Comment (kylef) [inline lib/cocoapods-core/podfile/dsl.rb]: @segiddins That's what I said "default subspecs"?

Comment (segiddins) [inline lib/cocoapods-core/podfile/dsl.rb]: ah must've misread it then

**Comment by kylef** [comment]
I've addressed the feedback. I'm still not completely happy with this, would appreciate any suggestions.

File: lib/cocoapods-core/podfile/dsl.rb [hunk 3]
```diff
@@ -112,6 +112,23 @@ module DSL
       #
       # ------
       #
+      # ### Subspecs
+      #
+      # When installing a Pod via it's name, it will install all of the
+      # default subspecs defined in the Pod.
```
Comment (segiddins) [inline lib/cocoapods-core/podfile/dsl.rb]: defined in the podspec?

File: lib/cocoapods-core/podfile/dsl.rb [hunk 4]
```diff
@@ -112,6 +112,23 @@ module DSL
...
+      #
+      # You may install a specific subspec using the following:
+      #
+      #     pod 'QueryKit/Attribute'
+      #
+      # You may specify a collection of subspecs to be installed as follows:
+      #
+      #     pod 'QueryKit', :subspecs => ['Attribute', 'QuerySet']
+      #
+      # *NOTE*: **Installing a subspec also installs the root spec for the pod.**
```
Comment (segiddins) [inline lib/cocoapods-core/podfile/dsl.rb]: That's only true if you use the `podspecs: ` shortcut, iirc

Comment (kylef) [inline lib/cocoapods-core/podfile/dsl.rb]: -> #236

Comment (segiddins) [inline lib/cocoapods-core/podfile/dsl.rb]: this is now no longer true

**Pull request merged**